src/meta/src/rpc/elections.rs

```diff
@@ -0,0 +1,838 @@
+        let (leader, _) = get_infos_obj(&mock_meta_store).await.unwrap();
+        assert!(
+            leader.eq(&test_leader),
+            "leader_info retrieved != leader_info send"
+        );
+    }
+
+    async fn put_lease_info<S: MetaStore>(lease: &MetaLeaseInfo, meta_store: &Arc<S>) {
+        let mut txn = Transaction::default();
+        txn.put(
+            META_CF_NAME.to_string(),
+            META_LEASE_KEY.as_bytes().to_vec(),
+            lease.encode_to_vec(),
+        );
+        if meta_store.txn(txn).await.is_err() {
+            panic!("Putting test lease failed");
+        }
+    }
+
+    async fn put_leader_info<S: MetaStore>(leader: &MetaLeaderInfo, meta_store: &Arc<S>) {
+        let mut txn = Transaction::default();
+        txn.put(
+            META_CF_NAME.to_string(),
+            META_LEADER_KEY.as_bytes().to_vec(),
+            leader.encode_to_vec(),
+        );
+        if meta_store.txn(txn).await.is_err() {
+            panic!("Putting test leader failed");
+        }
+    }
+
+    async fn put_leader_lease<S: MetaStore>(
+        leader: &MetaLeaderInfo,
+        lease: &MetaLeaseInfo,
+        meta_store: &Arc<S>,
+    ) {
+        put_leader_info(leader, meta_store).await;
+        put_lease_info(lease, meta_store).await;
+    }
+
+    /// Default setup
+    /// ## Returns:
+    /// lease timeout, meta store, leader info, lease info, lease registration time
+    async fn default_setup() -> (u64, Arc<MemStore>, MetaLeaderInfo, MetaLeaseInfo, Duration) {
+        let lease_timeout = 10;
+        let mock_meta_store = Arc::new(MemStore::new());
+        let leader_info = MetaLeaderInfo {
+            node_address: "localhost:1234".into(),
+            lease_id: 123,
+        };
+        let now = since_epoch();
+        let lease_info = MetaLeaseInfo {
+            leader: Some(leader_info.clone()),
+            lease_register_time: now.as_secs(),
+            lease_expire_time: now.as_secs() + lease_timeout,
+        };
+        put_leader_lease(&leader_info, &lease_info, &mock_meta_store).await;
+        (lease_timeout, mock_meta_store, leader_info, lease_info, now)
+    }
+
+    #[tokio::test]
+    async fn test_manage_term() {
+        let mock_meta_store = Arc::new(MemStore::new());
+        let lease_timeout = 10;
+
+        // Leader: If nobody was elected leader renewing lease fails and leader is marked as failed
+        let leader_info = MetaLeaderInfo {
+            node_address: "localhost:1234".into(),
+            lease_id: 123,
+        };
+        assert!(
+            !manage_term(true, &leader_info, lease_timeout, &mock_meta_store)
+                .await
+                .unwrap()
+        );
+
+        // Follower: If nobody was elected leader renewing lease also fails
+        assert!(
+            !manage_term(false, &leader_info, lease_timeout, &mock_meta_store)
+                .await
+                .unwrap()
+        );
```

Comment:
```suggestion
        assert_eq!(
            manage_term(false, &leader_info, lease_timeout, &mock_meta_store).await,
            Some(false)
        );
```

Reply:
Getting below error. Will keep `unwrap` for now

```
binary operation `==` cannot be applied to type `std::result::Result<bool, meta_store::MetaStoreError>
```

Reply:
I know. But even if I change it to `Ok(false)`, I get this error.